File: easy/fizzbuzz.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my ( $f, $b, $n ) = split( / /, $line );
    my @r;
    for ( my $i = 1 ; $i <= $n ; $i++ ) {
        @r[ $i - 1 ] =
          $i % $f == 0 ? $i % $b == 0 ? "FB" : "F" : $i % $b == 0 ? "B" : $i;
    }
    printf( "%s\n", join( ' ', @r ) );
}
close(INFILE);

File: easy/fizzbuzz.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split.map(&:to_i)
  (1..s[2]).each do |i|
    print ' ' if i > 1
    if i % s[0] > 0 && i % s[1] > 0
      print i
    else
      print 'F' if i % s[0] == 0
      print 'B' if i % s[1] == 0
    end
  end
  puts
end
